libkbfs/connection_status.go

```diff
@@ -50,7 +50,13 @@ func (kcs *kbfsCurrentStatus) PushConnectionStatusChange(service string, err err
 	defer kcs.lock.Unlock()
 
 	if err != nil {
+		// Exit early if the service is already failed, to avoid an
+		// invalidation.
+		_, errExisted := kcs.failingServices[service]
```

Comment:
Yes, the previous error has already invalidated it, but there's no risk of double close because after the invalidation, it immediately makes a new channel.  The reason to avoid the second invalidation (with the new channel) is to avoid the log spam that comes with it (and all subsequent, periodic invalidations).